### ISSUE TITLE
[Feature]: Exportable configuration

### ISSUE THREAD

**Issue by Printoiid** [wavetermdev/waveterm#1608]
### Feature description

It would be a nice to have to export my configuration and associated folders into a singular folder so that I can back up any backgrounds, themes, settings, and presets to GitHub so I can use it in the future if I need to reinstall my operating system, without worrying about anything being lot

### Implementation Suggestion

N/a, just an "export" command with WSH to a file path would be great

### Anything else?

n/a

**Comment by esimkowitz** [comment]
As a workaround, all your configs are stored in ~/.config/waveterm, should be easy enough to add to your dotfiles sync script.

**Comment by weilirs** [comment]
Hi, can I work on this?

**Comment by esimkowitz** [comment]
Sure, I'll assign to you. What's your planned implementation? I was thinking it could be cool to zip up the config dir and put it at a user-provided directory

**Comment by weilirs** [comment]
> ### Feature description
> It would be a nice to have to export my configuration and associated folders into a singular folder so that I can back up any backgrounds, themes, settings, and presets to GitHub so I can use it in the future if I need to reinstall my operating system, without worrying about anything being lot
> 
> ### Implementation Suggestion
> N/a, just an "export" command with WSH to a file path would be great

Yeah, I think this would be cool. I'll go with this approach then.

**Comment by weilirs** [comment]
If the PR looks good I'll update the documentation for this command.

**Comment by esimkowitz** [comment]
The PR looks good to me, made one small suggestion based on the CodeRabbit comment, but otherwise I think it's good to go!

**Comment by Lizardbutt** [comment]
Wouldn't it be a good idea to associate this Feature with the PR itself?

**Comment by Lizardbutt** [comment]
Alright, now we need another to import configs from a path. lmfao

**Comment by esimkowitz** [comment]
The logic there is a bit trickier, need to merge with existing configs and determine the override logic there
